homework-submission/react-hyf/react-hyf-client/src/App.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import './App.css';
+import Week1 from './components/week1';
+import Week2 from './components/week2';
+import Home from './components/main';
+import NavBar from './components/NavBar';
+import Week3 from './components/users';
+
+class App extends React.Component {
+  state = {
+    navElements: {
+      Home: { path: '/', description: '' },
+      'Week 1': { path: '/week1', description: 'Static Todolist' },
+      'Week 2': { path: '/week2', description: 'Dynamic Todolist' },
+      'Week 3': { path: '/week3', description: 'Lifecyle methods and fetch api' },
```

Comment:
I don't think `navElements` should be a state since we never update them.
you could just use `navElements = {...}` outside of state and in NavBar and Home component `navElements={this.navElements}`